### PR TITLE
Removing deprecated @Configuration annotation

### PR DESCRIPTION
**Pull request by krmahadevan**
Fixes # .

### Did you remember to?

- [ ] Add test case(s)
- [ ] Update `CHANGES.txt`

We encourage pull requests that:

* Add new features to TestNG (or)
* Fix bugs in TestNG

If your pull request involves fixing SonarQube issues then we would suggest that you please discuss this with the 
[TestNG-dev](https://groups.google.com/forum/#!forum/testng-dev) before you spend time working on it.

### CODE REVIEW

File: src/main/java/org/testng/internal/annotations/AnnotationHelper.java [hunk 1]
```diff
@@ -42,7 +42,6 @@
   };
 
   private static final Class[] CONFIGURATION_CLASSES = new Class[] {
```
Comment (juherr) [inline src/main/java/org/testng/internal/annotations/AnnotationHelper.java]: Is `Class<? extends IAnnotation>[]` working?

**Review by juherr — APPROVED** [review]
LGTM

File: src/main/java/org/testng/internal/annotations/AnnotationHelper.java (hunk 1 above)
Comment (krmahadevan) [reply src/main/java/org/testng/internal/annotations/AnnotationHelper.java]: You mean something like this ?

`private static final Class<? extends IAnnotation> CONFIGURATION_CLASSES = new Class[] {..}` ?

I think that should work.

Comment (juherr) [reply src/main/java/org/testng/internal/annotations/AnnotationHelper.java]: Yes, the goal is to avoid the cast later.

Comment (juherr) [reply src/main/java/org/testng/internal/annotations/AnnotationHelper.java]: Here: https://github.com/cbeust/testng/pull/1555/files#diff-8157237004c7fa80cfefa18fa2651745R292

Comment (krmahadevan) [reply src/main/java/org/testng/internal/annotations/AnnotationHelper.java]: Got it. Will take care once I am home.

Comment (krmahadevan) [reply src/main/java/org/testng/internal/annotations/AnnotationHelper.java]: Amended commit and fixed it.

File: src/main/java/org/testng/internal/Parameters.java [hunk 2]
```diff
@@ -150,23 +177,28 @@
         ? currentTestMethod.findMethodParameters(ctx.getCurrentXmlTest())
         : Collections.<String, String>emptyMap();
 
+    Class<? extends Annotation> annotation = retrieveConfigAnnotation(m);
+    String name = annotation == null ? "" : annotation.getSimpleName();
+    Class<? extends IAnnotation> annotationClass = null;
+    if (annotation != null) {
```
Comment (juherr) [inline src/main/java/org/testng/internal/Parameters.java]: null check on `annotation` made twice.

File: src/main/java/org/testng/internal/annotations/AnnotationHelper.java [hunk 3]
```diff
@@ -78,62 +77,33 @@ public static ITestAnnotation findTest(IAnnotationFinder finder, Constructor cto
...
-      IConfigurationAnnotation am = (IConfigurationAnnotation) finder.findAnnotation(ctor, IAfterMethod.class);
-
-      if (bs != null || as != null || bt != null || at != null || bg != null || ag != null
-          || bc != null || ac != null || bm != null || am != null)
-      {
-        result = createConfiguration(bs, as, bt, at, bg, ag, bc, ac, bm, am);
-      }
+  public static IConfigurationAnnotation findConfiguration(IAnnotationFinder finder, ConstructorOrMethod m) {
+    IConfigurationAnnotation result = null;
+    IConfigurationAnnotation bs = (IConfigurationAnnotation) finder.findAnnotation(m, IBeforeSuite.class);
```
Comment (juherr) [inline src/main/java/org/testng/internal/annotations/AnnotationHelper.java]: Is it possible to remove the cast?

File: src/main/java/org/testng/internal/annotations/AnnotationHelper.java [hunk 4]
```diff
@@ -78,62 +77,33 @@ public static ITestAnnotation findTest(IAnnotationFinder finder, Constructor cto
...
-      IConfigurationAnnotation bm = (IConfigurationAnnotation) finder.findAnnotation(ctor, IBeforeMethod.class);
-      IConfigurationAnnotation am = (IConfigurationAnnotation) finder.findAnnotation(ctor, IAfterMethod.class);
-
-      if (bs != null || as != null || bt != null || at != null || bg != null || ag != null
-          || bc != null || ac != null || bm != null || am != null)
-      {
-        result = createConfiguration(bs, as, bt, at, bg, ag, bc, ac, bm, am);
-      }
+  public static IConfigurationAnnotation findConfiguration(IAnnotationFinder finder, ConstructorOrMethod m) {
+    IConfigurationAnnotation result = null;
```
Comment (juherr) [inline src/main/java/org/testng/internal/annotations/AnnotationHelper.java]: I may be more readable to not have `result` and return values directly.

**Pull request merged**